cmd/manager/main.go

```diff
@@ -212,3 +226,22 @@ func setProfiler() {
 		}()
 	}
 }
+
+func lockHandler() (lockfile.Lockfile, error) {
+	lockFilePath, ok := os.LookupEnv("NMSTATE_INSTANCE_NODE_LOCK_FILE")
+	if !ok {
+		return "", errors.New("Failed to find NMSTATE_INSTANCE_NODE_LOCK_FILE ENV var")
+	}
+	log.Info(fmt.Sprintf("Try to take exclusive lock on file: %s", lockFilePath))
+	handlerLock, err := lockfile.New(lockFilePath)
+	if err != nil {
+		return handlerLock, errors.Wrapf(err, "failed to create lockFile for %s", lockFilePath)
+	}
+	err = wait.PollImmediate(5*time.Second, 600*time.Second, func() (done bool, err error) {
+		if handlerLock.TryLock() == nil {
+			return true, nil
+		}
+		return false, nil
```

Comment:
Can we handle this like normal golang err + if and print the error ?

```golang
err := handlerLock.TryLock()
if err != nil {
    log.Error(err, "retrying to lock handler")
    return false, nil // Don't return the error here, it will not re-poll if we do
}
return true, nil
```